src/main/java/seedu/meal360/Parser.java

```diff
@@ -0,0 +1,641 @@
+                    "Please Enter Additional Ingredients & Quantity (Enter done when complete): ");
+            while (true) {
+                String line = userInput.nextLine();
+                if (line.equals("done")) {
+                    ui.printSeparator();
+                    break;
+                } else {
+                    String[] command = line.trim().split(" ");
+                    ingredients = parseIngredientName(command);
+                    newIngredientList.putAll(ingredients);
+                }
+            }
+            recipeList.editRecipe(recipeToEdit, newIngredientList);
+        }
+
+        return recipeToEdit;
+    }
+
+    public String parseDeleteRecipe(String[] input, RecipeList recipeList) {
+        // user inputted recipe name
+        if (input[1].contains("/r")) {
+            // skip over /r in recipe name
+            String recipeToDelete = "";
+            for (int i = 2; i < input.length; i++) {
+                recipeToDelete += input[i] + " ";
+            }
+            recipeToDelete = recipeToDelete.trim();
+            if (recipeToDelete.equals("all")) {
+                String allRecipes = "";
+                int index = 0;
+                while (recipeList.size() != 0) {
+                    allRecipes += recipeList.deleteRecipe(index).getName() + ", ";
+                }
+                allRecipes = allRecipes.substring(0, allRecipes.length() - 2);
+                return allRecipes;
+            } else {
+                int recipeIndex = 0;
+                for (Recipe recipe : recipeList) {
+                    // find index of recipe we want to delete
+                    if (recipe.getName().equals(recipeToDelete)) {
+                        break;
+                    }
+                    recipeIndex++;
+                }
+                return recipeList.deleteRecipe(recipeIndex).getName();
+            }
+            // user inputted index of recipe in list
+        } else {
+            // deleting a range of recipes
+            if (input[1].length() >= 3) {
+                String rangeRecipes = "";
+                int startIndex = Integer.parseInt(input[1].charAt(0) + "");
+                int endIndex = Integer.parseInt(input[1].charAt(2) + "");
+                startIndex -= 1;
+                endIndex -= 1;
+                int newSize = recipeList.size() - ((endIndex - startIndex) + 1);
+                while (recipeList.size() != newSize) {
+                    rangeRecipes += recipeList.deleteRecipe(startIndex).getName() + ", ";
+                }
+                rangeRecipes = String.valueOf(new StringBuilder(rangeRecipes.substring(0, rangeRecipes.length() - 2)));
+                return rangeRecipes;
+            } else {
+                int recipeIndex = Integer.parseInt(input[1]);
+                // need to subtract 1 since list is 1-based
+                return recipeList.deleteRecipe(recipeIndex - 1).getName();
+            }
```

Comment:
Method seems quite big. Perhaps consider finding related behaviour and abstracting them into various functions